File: app/src/main/java/com/konstant/tool/httpsender/api/WeatherApi.java
package com.konstant.tool.httpsender.api;

import com.konstant.tool.httpsender.response.WeatherResponse;
import com.konstant.tool.httpsender.sender.Requester;

public interface WeatherApi {

    @Requester.Method(host = "http://tqapi.mobile.360.cn/", path = "v4/{path}.json", cache = true, methodType = Requester.MethodType.GET, response = WeatherResponse.class)
    void getWeatherInfo(@Requester.Path("path") String cityCode, Requester.Callback<?> callback);

}

File: tool/src/main/java/com/konstant/tool/lite/network/api/WeatherApi.kt
package com.konstant.tool.lite.network.api

import com.konstant.tool.lite.network.response.WeatherResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path

/**
* 作者：konstant
* 时间：2019/10/24 10:48
* 描述：360天气的接口
*/

interface WeatherApi {

    companion object {
        const val HOST = "http://tqapi.mobile.360.cn/"
    }

    @GET("v4/{path}.json")
    fun getWeatherWithCode(@Path("path") cityCode: String): Observable<WeatherResponse>

}